### ISSUE TITLE
Better type for getColumn

### ISSUE THREAD

**Issue by reverofevil** [SRombauts/SQLiteCpp#34]
It would be better to have `getColumn<T>(n)` syntax, because we need to set the type of the return value anyway. Writing `static_cast<T>(query.getColumn(n))` is way too verbose. You can achieve that by partial specialization of some class with static methods.

**Comment by HighSparrow** [comment]
Hi SRombauts,
code in style std::string value = query.getColumn(1); did works.
while code below doesn't:
std::string value = "";
value = query.getColumn(1);
Compiler(clang)'s issue is `Use of overloaded operator '=' is ambiguous (with operand types 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char> >') and 'SQLite::Column')`
I am trying to solve it, and I found something help in link: http://www.cplusplus.com/forum/general/28152/

**Comment by HighSparrow** [comment]
And you add method `inline operator const std::string() const` to fix an error required by GCC. I have no idea how to solve it now.

**Comment by worktycho** [comment]
Also with gcc and libstdc++ the following code fails as ambigous:
`int64_t id = query.getColumn(1);`
Because libstdc++ uses longs for fixed width types, whereas sqlite uses long longs.

**Comment by SRombauts** [comment]
Thanks for reporting, I'll look into this.

During this time, you could use the explicit forms like this:
```c++
std::string value;
value = query.getColumn(1).getText()
```
and
```c++
int64_t id = query.getColumn(1).getInt64();
```

Let me now if this suits you.
Or perhaps could you propose a new API?

**Comment by SRombauts** [comment]
I have reproduced your issues and will try to work on this in the following days, but I have not much spare times for this.

**Comment by SRombauts** [comment]
PB with 64bits long fixed for upcoming release 1.0.0

**Comment by SRombauts** [comment]
Closing this now due to age and inactivity; this API idea can be revisited if there is still demand.

**Comment by reverofevil** [comment]
Yeah, 12 years was enough time to invent vibeclosing, for C++ to die, and to quit C++ for good.
